verilog/analysis/verilog_linter.cc

```diff
@@ -205,28 +209,53 @@ std::vector<LintRuleStatus> VerilogLinter::ReportStatus(
   return statuses;
 }
 
-LinterConfiguration LinterConfigurationFromFlags() {
-  LinterConfiguration config;
-
-  // Turn on default ruleset.
-  const auto& ruleset = absl::GetFlag(FLAGS_ruleset);
-  config.UseRuleSet(ruleset);
+const std::string linter_config_file = ".rules.verible_lint";
 
+bool AppendLinterConfigurationFromFile(
+    LinterConfiguration &config, std::string config_filename) {
   // Read local configuration file
   std::string content;
-  if (verible::file::GetContents(absl::GetFlag(FLAGS_rules_config), &content)) {
+  if (verible::file::GetContents(config_filename, &content)) {
     RuleBundle local_rules_bundle;
     std::string error;
     if (local_rules_bundle.ParseConfiguration(content, &error)) {
       config.UseRuleBundle(local_rules_bundle);
     } else {
-      LOG(ERROR) << "Unable to fully parse configuration: " << error
-                 << std::endl;
+      LOG(WARNING) << "Unable to fully parse configuration: " << error;
+    }
+    return true;
+  }
+
+  return false;
+}
+
+LinterConfiguration LinterConfigurationFromFlags(absl::string_view filename) {
+  LinterConfiguration config;
+
+  // Turn on default ruleset.
+  const auto& ruleset = absl::GetFlag(FLAGS_ruleset);
+  config.UseRuleSet(ruleset);
+
+  if (FLAGS_rules_config.IsModified()) {
+    // Use configuration file from flag if specified
+    std::string config_file = absl::GetFlag(FLAGS_rules_config);
+
+    if (!AppendLinterConfigurationFromFile(config, config_file)) {
+      LOG(WARNING) << "Unable to open linter configuration file: "
+                   << config_file;
+    }
+  } else if (absl::GetFlag(FLAGS_rules_config_search)) {
+    // Search upward if search is enabled and no configuration file is
+    // specified
+    std::string config_file;
+    if (verible::file::UpwardFileSearch(std::string(filename),
+                                        linter_config_file,
+                                        config_file)) {
+      if (!AppendLinterConfigurationFromFile(config, config_file)) {
```

Comment:
I really think we're missing `--verbose` and `LOG(VERBOSE)` - this would be useful to log things like the file that is found and used here.

Reply:
We do have `VLOG` (frequently used in the codebase).

Reply:
@fangism but `VLOG` still does not have any `verbose` severity, moreover it appears to work exactly the same as regular `LOG`. Which one should be used then?

Should I add logging the config file used with `[V]LOG(INFO)` then? It seems redundant to log it each time, but I think it would be useful if there was a `--verbose` switch.

Reply:
With both `LOG` and `VLOG` messages get print to stderr if you run with `--alsologtostderr`.
When you use `VLOG(N)`, messages are controlled with `-v=M` for `M >= N`.
I'd recommend `VLOG(1)` to surface these messages with minimum verbosity.  Does that help?

Reply:
Thanks!

But it's not really what I meant. `VLOG(1)`  is basically `INFO`, this would be printed all the time with the default settings. I think that the information about the used config file should not be visible by default, unless the user asks for it. The `-v=X` flag also doesn't seem to work for me. Should it be available for linter?

Reply:
@hzeller is there a behavioral difference between internal/external logging?

Reply:
Yes, right  now the equivalent of -v=X is setting the environment variable `TF_CPP_VMODULE` (why TF ? well, I think this is because i copied it from tensor-flow and missed that reference. Probably should rename that to `CPP_VMODULE`)